### PR TITLE
[Bugfix:Forum] Prevent img markdown on the forum

### PR DESCRIPTION
**Pull request by andrewaikens87**
### Please check if the PR fulfills these requirements:

* [x] The PR title and message follows our [guidelines](https://submitty.org/developer/how_to_contribute)
* [x] Tests for the changes have been added/updated (if possible)
* [x] Documentation has been updated/added if relevant

### What is the current behavior?
Users can post inline images. 

### What is the new behavior?
Regex to convert users inline attachments to links.

### CODE REVIEW

**Comment by MasterOdin** [comment]
Looking at this and https://github.com/Submitty/submitty.github.io/pull/146, it's not clear to me (as an outsider in the development of the forum) why this is happening, and I think it's important that gets documented, especially given that it's non-standard markdown feature that anyone would probably expect to work out of the box.

Additionally, some possible cases here that I think will be confusing without knowing of the behavior beforehand:
1. hyperlinked images
`[](https://example.com)` will be rewritten (I believe) to be `[http://underthebridge.co.uk/wp-content/uploads/2014/03/Example-main-image1.jpg](https://example.com)` and then show as a link, but be hyperlinked to somewhere else. This could be confusing.
2. images with titles
Test `Alt Text Test` will be rewritten to `Test http://underthebridge.co.uk/wp-content/uploads/2014/03/Example-main-image1.jpg "Example Title" Test` which feels like it could look odd.

**Comment by andrewaikens87** [comment]
This only takes the url from the image and adds it in as a hyperlink. When a user submits a link in plaintext the engine will convert it to a hyperlink. This regex uses this feature. We prefer users upload images as attachments.

**Comment by MasterOdin** [comment]
Also, from a cursory look again, I think you're going to have to repeatedly call the preg_replace on a post until it does not actually replace anything. For example, currently, an image could be inlined by doing: `)` (repl example: https://repl.it/repls/RunnyPureCategories).

**Comment by andrewaikens87** [comment]
Fixed!

File: site/app/controllers/forum/ForumController.php [hunk 1]
```diff
@@ -786,6 +786,12 @@ public function showThreads($thread_id = null, $option = 'tree'){
         if(!empty($posts)){
             $thread_id = $posts[0]["thread_id"];
         }
+        foreach($posts as &$post) {
+            $count = 1;
+            while($count > 0) {
+                $post['content'] = preg_replace('/(?:!\[(.*?)\]\((.*?)\))/', '$2', $post['content'], -1, $count);
+            }
+        }
```
Comment (MasterOdin) [inline site/app/controllers/forum/ForumController.php]: Use the little thought of do-while here for a bit more readability in the sequence of what you're doing.
```
do {
    $post['content'] = preg_replace('/(?:!\[(.*?)\]\((.*?)\))/', '$2', $post['content'], -1, $count);
} while ($count > 0);
```

Comment (andrewaikens87) [reply site/app/controllers/forum/ForumController.php]: fixed

**Pull request merged**